src/Test/Utilities/Portable/Compilation/ControlFlowGraphVerifier.cs

```diff
@@ -1012,17 +1012,23 @@ bool isLongLivedCaptureReferenceSyntax(SyntaxNode captureReferenceSyntax)
                     case LanguageNames.CSharp:
                         {
                             var syntax = (CSharpSyntaxNode)captureReferenceSyntax;
-                            switch (syntax.Kind())
+                            while (syntax is CSharp.Syntax.ExpressionSyntax)
                             {
-                                case CSharp.SyntaxKind.ObjectCreationExpression:
-                                    if (((CSharp.Syntax.ObjectCreationExpressionSyntax)syntax).Initializer?.Expressions.Any() == true)
-                                    {
-                                        return true;
-                                    }
-                                    break;
+                                switch (syntax.Kind())
+                                {
+                                    case CSharp.SyntaxKind.ObjectCreationExpression:
+                                        if (((CSharp.Syntax.ObjectCreationExpressionSyntax)syntax).Initializer?.Expressions.Any() == true)
+                                        {
+                                            return true;
+                                        }
+                                        break;
+                                    default:
+                                        syntax = syntax.Parent;
```

Comment:
>syntax = syntax.Parent [](start = 40, length = 22)

This looks too relaxed. Basically any syntax below CSharp.Syntax.ExpressionSyntax will return true. What scenario are we trying to handle?

Reply:
We're trying to handle this scenario: [https://github.com/dotnet/roslyn/pull/31724/files#diff-44ef19e4c61837033de3e3b7f6002830R12798](https://github.com/dotnet/roslyn/pull/31724/files#diff-44ef19e4c61837033de3e3b7f6002830R12798). Where the property references uses the flow capture from a containing region.

---
In reply to: [241497757](https://github.com/dotnet/roslyn/pull/31724#discussion_r241497757) [](ancestors = 241497757)